lib/main.rb

```diff
@@ -1,27 +1,30 @@
 require_relative "./product"
-require_relative "./warehouse"
 require_relative "./basket"
 
-p1 = Product.new(name: "Football", price: 79)
-p2 = Product.new(name: "Basketball", price: 60)
-p3 = Product.new(name: "Squash racket type A", price: 150)
-p4 = Product.new(name: "Squash racket type B", price: 185.60)
-p5 = Product.new(name: "Sports bag", price: 145)
+begin
+  p1 = Product.new(name: "Football", price: 79.44)
+  p2 = Product.new(name: "Basketball", price: 60)
+  p3 = Product.new(name: "Squash racket type A", price: 150)
+  p4 = Product.new(name: "Squash racket type B", price: 185.60)
+  p5 = Product.new(name: "Sports bag", price: 145)
 
-basket = Basket.new
+  basket = Basket.new
 
-basket.add(p1)
-basket.add(p2)
-basket.add(p2)
-basket.add(p2)
-basket.add(p3)
+  basket.add(p1)
+  basket.add(p2)
+  basket.add(p2)
+  basket.add(p2)
+  basket.add(p3)
 
-p basket.show
+  p basket.show
 
-basket.delete(10)
-basket.delete(2)
-
-p basket.show
+  basket.delete(2)
+  p basket.show
 
+  basket.delete(0)
 
+rescue
+  Basket::ProductNotFound
```

Comment:
it should be a single line `rescue Foo` or `rescue Foo => exception` in more complex situations.